### ISSUE TITLE
gitStream not working in one of repository in our organization, but it works in my personal repository.

### ISSUE THREAD

**Issue by Sekhar9581** [linear-b/gitstream#188]
**Describe the bug**
First, I have tried GITStream with few automation actions in my personal GitHub repository, it worked fine. But same not working in one of repository of our organization. I have installed GITStream in a repository and added gitstream.cm and gitstream.yml file properly, still I cannot see any action running. Tried with different automations, nothing works. Do I need to focus on any organization/repository settings?

**To Reproduce**

1. The `.cm` automation file
2. The PR URL (as it contains the repo and PR identifiers)
3. Describe your PR relevant content 
4. Add relevant commit SHA

**Expected behavior**

**Screenshots**
<img width="611" alt="gitstream" src="https://github.com/linear-b/gitstream/assets/111882050/f2975d0f-6030-4c7c-8a50-b956c012cd4a">

**Additional context**

**Comment by vim-zz** [comment]
@Sekhar9581 Thank you for giving gitStream a try!

I suggets to check the following:

Go to https://github.com/apps/gitstream-cm/installations/new and check that under `Repository access` you selected either `All` or the repositories you want gitStream to work in.

Check you see the following directory strcuture in your repo
```
.
├─ .cm/
│  └─ gitstream.cm
├─ .github/
│  └─ workflows/
│     └─ gitstream.yml
```

**Comment by Sekhar9581** [comment]
Thanks for responding. Verified that I have selected only one repository for access and I can see gitStream under Settings>Github Apps and gitstream.cm and gitstream.yml are placed in proper directory. I cannot see any workflows in Actions tab.
<img width="700" alt="gitstream" src="https://github.com/linear-b/gitstream/assets/111882050/383af730-4889-426e-9cc8-8d5fca97efbb">
<img width="576" alt="gitstream" src="https://github.com/linear-b/gitstream/assets/111882050/2befcb9a-9f74-496e-a159-440cd110a8f0">

**Comment by vim-zz** [comment]
Does both files merged to the default branch?

```
.
├─ .cm/
│  └─ gitstream.cm
├─ .github/
│  └─ workflows/
│     └─ gitstream.yml
```

**Comment by Sekhar9581** [comment]
Yes. I have main(default) and test branches, both are having the cm and yml files. Im making changes in test branch and trying to merge into main through pull request. I cannot see gitstream running as shown in below snap
<img width="548" alt="gitstream" src="https://github.com/linear-b/gitstream/assets/111882050/99acf20e-67bb-407a-b7ef-0f722aa16bf3">

**Comment by vim-zz** [comment]
It seems that gitStream service isn't triggered by GitHub on your pull request, therefore it doesn't run the GitHub action at all. Can you share the org name you installed gitStream for (it's the 2nd element in the PR URL, right after https://github.com)? It will help to make sure that it was installed properly

**Comment by vim-zz** [comment]
@Sekhar9581 we see an issue with gitStream access to your org. Please add gitStream app to your authorized IPs, see GitHub instruction: https://docs.github.com/en/enterprise-cloud@latest/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/managing-allowed-ip-addresses-for-your-organization#allowing-access-by-github-apps

**Comment by Sekhar9581** [comment]
Hi @vim-zz took some time to get back to you. Before enabling IP allow list for installed GitHub Apps, I have some queries. Can you please help me to understand below?

1. How do I check the IP address of gitstream before enabling IP allow list?
2. We are in a restricted environment and only allowed internal IPs for access. Is it safe to allow gitstream IP in our environment?
3. Will gitstream expose my code to the outside world? If yes, what level of code will be exposed externally?

**Comment by vim-zz** [comment]
Hi @Sekhar9581 

Since gitStream is a GitHub app, you can automatically add to your allow list the IP addresses configured for GitHub apps that you install in your organization [learn more here](https://docs.github.com/en/enterprise-cloud@latest/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/managing-allowed-ip-addresses-for-your-organization#allowing-access-by-github-apps)
2. In this configuration, you allow to use gitStream app IP only to react on events that originated from GitHub while not allowing any other traffic from the IP of gitStream app
3. gitStream doesn’t extract your source code. Like any other CI/CD automation, the source code is being scanned in the repo using your CI runners and is not shared with any external services

I hope it clarify things, let me know if you have more questions

**Comment by Sekhar9581** [comment]
Hi @vim-zz 
Thanks for clarifying the queries. We have enabled IP allow list for installed GitHub Apps in our Organization. After enable IP allow list, I created new pull request. gitStream ran in the background while I was creating the pull request, but after sometime it showed as checks skipped. I have attached the screenshot. Please let me know if I missed anything.
Thanks.
<img width="580" alt="gitStream" src="https://github.com/linear-b/gitstream/assets/111882050/cef9a770-88ba-4f32-8ce7-6d98d1d64d71">

**Comment by BenLloydPearson** [comment]
Hi @Sekhar9581 thanks for your patience with us.

Are you still having this issue? If so, do you see any errors in your GitHub Actions log for this?

**Comment by vim-zz** [comment]
not active any more, please re-open if you encounter it
